app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/complete_ruling_section.scala.html

```diff
@@ -0,0 +1,43 @@
+@*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import uk.gov.hmrc.tariffclassificationfrontend.models.{Case, CaseStatus}
+@import uk.gov.hmrc.tariffclassificationfrontend.forms.DecisionFormData
+@(c: Case, decisionForm : Option[Form[DecisionFormData]], classes: Seq[String] = Seq.empty)(implicit messages: Messages)
+
+@if(c.decision.isDefined && c.status == CaseStatus.OPEN) {
+
+    <div id="complete-case-section">
+        @if(decisionForm.map(_.errors.size).exists(_ > 0)) {
+            <h2 class="heading-medium">@messages("case.decision.before-you-complete")</h2>
+            <p>To complete this case, you need to fill in these fields:</p>
+            <ul class="list list-bullet">
+            @for(error <- decisionForm.get.errors) {
+                <li><span id="constrain-@error.key">@messages("case.decision.constrain." + error.key)</span></li>
+            }
+            </ul>
+
+            <a id="complete-case-button-disabled" disabled="disabled" class="button @classes.mkString(" ")" >
```

Comment:
You can do `@if(decisionForm.map(_.errors.size).exists(_ > 0)){disabled="disabled"}` rather than having two of the same buttons

Reply:
Is if possible to extract `decisionForm.map(_.errors.size).exists(_ > 0)` into a boolean field and use it in both if tests?

Reply:
Is not just the disable is as well the href and all the content.  Do think is worth it?

Reply:
Yep you can use `@defining(expression) { value => .... some html... }`

Reply:
Disabled disabled the href @pllaurado